### ISSUE TITLE
Bind key to cleanly toggle line wrapping

### ISSUE THREAD

**Issue by bew** [gwsw/less#272]
Hello,
I usually open files with less to explore them and by default the long lines are auto-wrapping (unless we do horizontal movements).
Sometimes I want to wrap the lines temporarily (to show overall structure of the file to someone, and them show a specific line, etc...).

I want to be able to toggle wrapping on demand with Alt-w.
I currently have:
_(note: that `toggle-option` action doesn't seems to be documented anywhere)_
```
#command
\ew toggle-option --chop-long-lines\n\n
```
However this isn't great:
1. Option toggling seems to require an explicit action from me (press enter) and doesn't work right away.
   :point_right: I noticed that triggering this keybind twice does toggle the option, not exactly sure why.
2. The scrollback moves a lot when toggling long lines wrapping, going up by half a page sometimes (when I'm looking at the middle/bottom of a file), making it really hard to follow where I was before the option toggle.

For (1), I would expect to only have to trigger the keybind once, and have the option toggled.

For (2), I would expect the current top line to remain at the same place, and only the visible lines should be affected by the wrapping change.
I think that remembering the top line before the option change and move to that top line after the change should work to that end, but I don't think I can express that in the lesskey config file.

---
What do you think ?
_Are there some (old) terminal limitations I'm not seeing that would prevent this to be possible?_

**Comment by gwsw** [comment]
The toggle-option command (documented in the lesskey man page) invokes the toggle-option functionality, so the first dash in the extra string is not needed. I think the extra dash is causing most of your problems. Also only one newline is needed at the end. So the lesskey line should be
```
\ew toggle-option -chop-long-lines\n
```
You could also do it more simply using the single-letter alias -S, which doesn't need the extra dash or the newline:
```
\ew toggle-option S
```
In either case you can suppress the status message at the bottom of the screen by adding a ^P to the beginning of the extra string.

**Comment by bew** [comment]
```
\ew toggle-option ^P-chop-long-lines\n
```
seems to work as intended :tada:
Thank you!! :)

> documented in the lesskey man page

In the man page of lesskey, doing a search for `toggle` only gives the follwing 3 lines:
```
            -          toggle-option
            :t         toggle-option t
            s          toggle-option o
```
Is that the documentation you're talking about ?
I think it would help to have a little bit more doc, for example to say how to give the option (in short or long form), and mention the ^P trick you're talking about

(I'm on less v600)

**Comment by gwsw** [comment]
You're right, the documentation is a bit sparse. To avoid duplication, lesskey merely documents the name of the command; the command itself is documented in the less man page, which is where ^P is described. But the need to use exactly one dash for long option names after toggle-option isn't really documented anywhere. I will look into improving the docs in this area.

**Comment by bew** [comment]
> the command itself is documented in the less man page

FYI In the man page of `less`, searching for `toggle` doesn't yield any match

**Comment by gwsw** [comment]
Right, the command is documented as "-", the key that invokes it. Less doesn't list the lesskey names for each command, since most less users won't ever edit a lesskey file. That does make it harder to associate the command names in the lesskey man page with the commands in the less man page.

**Comment by bew** [comment]
oh! okay, yeah I know I'm in the minority here, wanting custom keybinds for everything ^^

Do as you can :)
